aclk/aclk.c

```diff
@@ -240,6 +241,44 @@ static void msg_callback(const char *topic, const void *msg, size_t msglen, int
     aclk_handle_cloud_message(cmsg);
 }
 
+static void msg_callback_new(const char *topic, const void *msg, size_t msglen, int qos)
+{
+    if (msglen > RX_MSGLEN_MAX)
+        error("Incoming ACLK message was bigger than MAX of %d and got truncated.", RX_MSGLEN_MAX);
+
+    debug(D_ACLK, "Got Message From Broker Topic \"%s\" QOS %d", topic, qos);
+
+    if (aclk_shared_state.mqtt_shutdown_msg_id > 0) {
+        error("Link is shutting down. Ignoring message.");
+        return;
+    }
+
+    const char *msgtype = strrchr(topic, '/');
+    if (unlikely(!msgtype)) {
+        error_report("Cannot get message type from topic. Ignoring message from topic \"%s\"", topic);
+        return;
+    }
+    msgtype++;
+    if (unlikely(!*msgtype)) {
+        error_report("Message type empty. Ignoring message from topic \"%s\"", topic);
+        return;
+    }
+
+#ifdef ACLK_LOG_CONVERSATION_DIR
+#define FN_MAX_LEN 512
+    char filename[FN_MAX_LEN];
+    int logfd;
+    snprintf(filename, FN_MAX_LEN, ACLK_LOG_CONVERSATION_DIR "/%010d-rx-%s.bin", ACLK_GET_CONV_LOG_NEXT(), msgtype);
```

Comment:
Please, add one character for `filename` or subtract one byte here inside the argument. I prefer the first option.

Reply:
`The functions snprintf() and vsnprintf() write at most size bytes (including the terminating null byte ('\0')) to str.`

```
int snprintf ( char * s, size_t n, const char * format, ... );
Write formatted output to sized buffer
Composes a string with the same text that would be printed if format was used on printf, but instead of being printed, the content is stored as a C string in the buffer pointed by s (taking n as the maximum buffer capacity to fill).

If the resulting string would be longer than n-1 characters, the remaining characters are discarded and not stored, but counted for the value returned by the function.

A terminating null character is automatically appended after the content written.

After the format parameter, the function expects at least as many additional arguments as needed for format.
```

if you do program like following:
```C
        char buf[5];
        snprintf(buf, 5, "0123456789");
```
and open it in GDB  you can see result is properly terminated string "0123" not exceeding 5 bytes

Reply:
I called your attention for this, because we are following this pattern on other files like [this](https://github.com/netdata/netdata/blob/37bee1d197229a4de1d6eac9942fc626f1da8ee3/collectors/proc.plugin/plugin_proc.c#L248-L250).

The unique real problem that we will have with this, it is to have two different patterns inside our code.

Reply:
I prefer correctness over patterns. Adding byte here will just add useless byte of memory and confusion for whoever reads the code.

Reply:
You are right, this is not a source of bug. 

I will consider that it is not possible to reach 512 bytes and I will start the tests of this PR now.